src/application/ui/appContainer/actions/handlers.ts

```
@@ -22,3 +25,15 @@ export function* showErrorDialogAsync(dialog: {
     }
   ));
 }
+
+/**
+ * Async action handler: Navigate to route
+ * @param option Route name and params
+ */
+export function* navigateAsync(option: NavigationNavigateActionPayload) {
+  const state = (yield select()) as StoreState;
+  if (state.ui.appContainer.navigator) {
+    state.ui.appContainer.navigator.dispatch(
+      NavigationActions.navigate(option));
+  } else throwError('application_improper', 'Application is in improper state');
```

Unexpected should be throw using normal throw statement `throw new Error('Application navigator is not set!')`. `throwError` method is for checked error.